[Sitemaps] limit on "bad url" log messages

If a sitemap is erroneously detected as plain-text sitemap (cf. #144), SiteMapParser may report all or most of the file content as "bad url". This may result
- either in many log messages:
```
2016-12-22 13:55:26.628 c.s.SiteMapParser [WARN] Bad url: [<?xml version="1.0" encoding="UTF-8"?>]
2016-12-22 13:55:26.628 c.s.SiteMapParser [WARN] Bad url: [<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:news="http://www.google.com/schemas/sitemap-news/0.9">]
2016-12-22 13:55:26.628 c.s.SiteMapParser [WARN] Bad url: [                <url>]
2016-12-22 13:55:26.628 c.s.SiteMapParser [WARN] Bad url: [        <loc>http://www.azonline.de/Sport/Fussball/1.-Bundesliga/2640159-Hamburger-SV-Bruchhagen-Lasse-mich-bei-der-Sportchef-Suche-nicht-hetzen</loc>]
2016-12-22 13:55:26.628 c.s.SiteMapParser [WARN] Bad url: [        <news:news>]
2016-12-22 13:55:26.628 c.s.SiteMapParser [WARN] Bad url: [            <news:publication>]
2016-12-22 13:55:26.628 c.s.SiteMapParser [WARN] Bad url: [                <news:name>Allgemeine Zeitung</news:name>]
2016-12-22 13:55:26.628 c.s.SiteMapParser [WARN] Bad url: [                <news:language>ger</news:language>]
2016-12-22 13:55:26.628 c.s.SiteMapParser [WARN] Bad url: [            </news:publication>]
2016-12-22 13:55:26.628 c.s.SiteMapParser [WARN] Bad url: [            <news:publication_date>2016-12-22T14:52:00Z</news:publication_date>]
2016-12-22 13:55:26.628 c.s.SiteMapParser [WARN] Bad url: [            <news:title>Hamburger SV : Bruchhagen: Lasse mich bei der Sportchef-Suche nicht hetzen</news:title>]
2016-12-22 13:55:26.628 c.s.SiteMapParser [WARN] Bad url: [        </news:news>]
2016-12-22 13:55:26.628 c.s.SiteMapParser [WARN] Bad url: [    </url>]
2016-12-22 13:55:26.628 c.s.SiteMapParser [WARN] Bad url: [                        <url>]
... (5000 lines following)
```
- or even in one very long message (more than 180 kB in a single line):
```
2016-12-22 14:43:22.173 c.s.SiteMapParser [WARN] Bad url: [<?xml version="1.0" encoding="UTF-8"?> <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:n="http://www.google.com/schemas/sitemap-news/0.9"> <url> <loc>http://www.hotnews.ro/stiri-politic-21489493-liviu-dragnea-cer-public-serviciilor-secrete-spuna-daca-exista-vreo-problema-securitate-legata-sotul-premierului-propus-sevil-shhaideh.htm</loc> <n:news> <n:publication> <n:name>HotNews.ro</n:name> <n:language>ro</n:language> </n:publication> <n:genres>PressRelease</n:genres> <publication_date>2016-12-22T16:16:35</publication_date> <n:title><![CDATA[Liviu Dragnea: Cer public serviciilor secrete sa spuna daca exista vreo problema de securitate legata de sotul premierului propus Sevil Shhaideh]]></n:title> <n:keywords><![CDATA[]]></n:keywords> </n:news> </url> ...
```
There should be a limit on both the max. number of lines and the line length, logged as error. This avoids consequential errors, e.g.:
```
2016-12-22 14:43:22,176 ERROR Unable to write to stream UDP:localhost:514 for appender syslog
2016-12-22 14:43:22,176 ERROR An exception occurred processing Appender syslog org.apache.logging.log4j.core.appender.AppenderLoggingException: Error flushing stream UDP:localhost:514
...
Caused by: java.io.IOException: Message too long (sendto failed)
```

--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -1,6 +1,7 @@
 Crawler-Commons Change Log
 
 Current Development 0.10-SNAPSHOT (yyyy-mm-dd)
+- [Sitemaps] Limit on "bad url" log messages (sebastian-nagel) #145
 - EffectiveTldFinder to parse Internationalized Domain Names (sebastian-nagel) #179
 - Add main() to EffectiveTldFinder (sebastian-nagel) #187
 - Handle new suffixes in PaidLevelDomain (kkrugler) #183

--- a/src/main/java/crawlercommons/sitemaps/SiteMapParser.java
+++ b/src/main/java/crawlercommons/sitemaps/SiteMapParser.java
@@ -298,10 +298,23 @@ public class SiteMapParser {
         BufferedReader reader = new BufferedReader(new InputStreamReader(bomIs, UTF_8));
 
         String line;
-        int i = 1;
-        while ((line = reader.readLine()) != null) {
-            if (line.length() > 0 && i <= MAX_URLS) {
-                addUrlIntoSitemap(line, textSiteMap, null, null, null, i++);
+        int i = 0;
+        while ((line = reader.readLine()) != null && ++i <= MAX_URLS) {
+            line = line.trim();
+            if (line.isEmpty())
+                continue;
+            try {
+                URL url = new URL(line);
+                boolean valid = urlIsValid(textSiteMap.getBaseUrl(), url.toString());
+                if (valid || !strict) {
+                    SiteMapURL sUrl = new SiteMapURL(url, valid);
+                    textSiteMap.addSiteMapUrl(sUrl);
+                    LOG.debug("  {}. {}", i, sUrl);
+                } else {
+                    LOG.debug("URL: {} is excluded from the sitemap as it is not a valid url = not under the base url: {}", url.toExternalForm(), textSiteMap.getBaseUrl());
+                }
+            } catch (MalformedURLException e) {
+                LOG.debug("Bad url: [{}]", line.substring(0, Math.min(1024, line.length())));
             }
         }
         textSiteMap.setProcessed(true);
@@ -407,42 +420,6 @@ public class SiteMapParser {
             }
         } catch (ParserConfigurationException e) {
             throw new IllegalStateException(e);
-        }
-    }
-
-    /**
-     * Adds the given URL to the given sitemap while showing the relevant logs
-     * 
-     * @param urlStr
-     *            an URL string to add to the
-     *            {@link crawlercommons.sitemaps.SiteMap}
-     * @param siteMap
-     *            the sitemap to add URL(s) to
-     * @param lastMod
-     *            last time the {@link crawlercommons.sitemaps.SiteMapURL} was
-     *            modified
-     * @param changeFreq
-     *            the {@link crawlercommons.sitemaps.SiteMapURL} change frquency
-     * @param priority
-     *            priority of this {@link crawlercommons.sitemaps.SiteMapURL}
-     * @param urlIndex
-     *            index position to which this entry has been added
-     */
-    protected void addUrlIntoSitemap(String urlStr, SiteMap siteMap, String lastMod, String changeFreq, String priority, int urlIndex) {
-        try {
-            URL url = new URL(urlStr); // Checking the URL
-            boolean valid = urlIsValid(siteMap.getBaseUrl(), url.toString());
-
-            if (valid || !strict) {
-                SiteMapURL sUrl = new SiteMapURL(url.toString(), lastMod, changeFreq, priority, valid);
-                siteMap.addSiteMapUrl(sUrl);
-                LOG.debug("  {}. {}", urlIndex + 1, sUrl);
-            } else {
-                LOG.warn("URL: {} is excluded from the sitemap as it is not a valid url = not under the base url: {}", url.toExternalForm(), siteMap.getBaseUrl());
-            }
-        } catch (MalformedURLException e) {
-            LOG.warn("Bad url: [{}]", urlStr);
-            LOG.trace("Can't create a sitemap entry with a bad URL", e);
         }
     }
 

--- a/src/main/java/crawlercommons/sitemaps/SiteMapParser.java
+++ b/src/main/java/crawlercommons/sitemaps/SiteMapParser.java
@@ -301,8 +301,9 @@ public class SiteMapParser {
         int i = 0;
         while ((line = reader.readLine()) != null && ++i <= MAX_URLS) {
             line = line.trim();
-            if (line.isEmpty())
+            if (line.isEmpty()) {
                 continue;
+            }
             try {
                 URL url = new URL(line);
                 boolean valid = urlIsValid(textSiteMap.getBaseUrl(), url.toString());

